### PR TITLE
Input capture documentation

### PR DESCRIPTION
**Pull request by feschber**

### CODE REVIEW

**Comment by feschber** [comment]
FYI: I added the examples over at [ashpd-mre](https://github.com/feschber/ashpd-mre) to test them.

**Comment by feschber** [comment]
I hope it is okay to have `reis` as a `dev-dependency`. (otherwise the tests fail)

File: src/desktop/input_capture.rs [hunk 1]
```diff
@@ -1,3 +1,269 @@
...
+//!     let (session, capabilities) = input_capture
+//!         .create_session(
+//!             &ashpd::WindowIdentifier::default(),
+//!             Capabilities::Keyboard | Capabilities::Pointer | Capabilities::Touchscreen,
+//!         )
+//!         .await?;
+//!     eprintln!("capabilities: {capabilities}");
+//!
+//!     let eifd = input_capture.connect_to_eis(&session).await?;
+//!     eprintln!("        eifd: {}", eifd.as_raw_fd());
```
Comment (bilelmoussaoui) [inline src/desktop/input_capture.rs:28]:
```suggestion
//!     eprintln!("eifd: {}", eifd.as_raw_fd());
```

File: src/desktop/input_capture.rs [hunk 2]
```diff
@@ -1,3 +1,269 @@
...
+//!     Left,
+//!     Right,
+//!     Top,
+//!     Bottom,
+//! }
+//!
+//! static INTERFACES: OnceLock<HashMap<&'static str, u32>> = OnceLock::new();
+//!
+//! async fn run() -> ashpd::Result<()> {
+//!     let input_capture = InputCapture::new().await.unwrap();
```
Comment (bilelmoussaoui) [inline src/desktop/input_capture.rs:138]:
```suggestion
//!     let input_capture = InputCapture::new().await?;
```

File: src/desktop/input_capture.rs [hunk 3]
```diff
@@ -1,3 +1,269 @@
...
+//!
+//!     let (session, _cap) = input_capture
+//!         .create_session(
+//!             &ashpd::WindowIdentifier::default(),
+//!             Capabilities::Keyboard | Capabilities::Pointer | Capabilities::Touchscreen,
+//!         )
+//!         .await?;
+//!
+//!     // connect to eis server
+//!     eprintln!("connect_to_eis");
```
Comment (bilelmoussaoui) [inline src/desktop/input_capture.rs:148]:
```suggestion
```

File: src/desktop/input_capture.rs [hunk 4]
```diff
@@ -1,3 +1,269 @@
...
+//!         m.insert("ei_pingpong", 1);
+//!         m.insert("ei_seat", 1);
+//!         m.insert("ei_device", 2);
+//!         m.insert("ei_pointer", 1);
+//!         m.insert("ei_pointer_absolute", 1);
+//!         m.insert("ei_scroll", 1);
+//!         m.insert("ei_button", 1);
+//!         m.insert("ei_keyboard", 1);
+//!         m.insert("ei_touchscreen", 1);
+//!         m
```
Comment (bilelmoussaoui) [inline src/desktop/input_capture.rs:173]: Maybe using `HashMap::from(&[])` would make the code slightly nicer?

File: src/desktop/input_capture.rs [hunk 5]
```diff
@@ -1,3 +1,269 @@
...
+//!     let failed_barrier_ids = response.failed_barriers();
+//!
+//!     eprintln!("failed barrier ids: {:?}", failed_barrier_ids);
+//!
+//!     input_capture.enable(&session).await?;
+//!
+//!     let mut activate_stream = input_capture.receive_activated().await?;
+//!
+//!     loop {
+//!         let activated = activate_stream.next().await.unwrap();
```
Comment (bilelmoussaoui) [inline src/desktop/input_capture.rs:222]:
```suggestion
//!         let activated = activate_stream.next().await?;
```

**Review by bilelmoussaoui — COMMENTED** [review]
otherwise lgtm, thanks a lot!

**Comment by bilelmoussaoui** [comment]
> I hope it is okay to have `reis` as a `dev-dependency`. (otherwise the tests fail)

yep, no problem with that :)

File: src/desktop/input_capture.rs (hunk 4 above)
Comment (feschber) [reply src/desktop/input_capture.rs]: done

File: src/desktop/input_capture.rs (hunk 5 above)
Comment (feschber) [reply src/desktop/input_capture.rs:221]: this does not work, stream returns an option and I could otherwise not return ashpd::Result

**Comment by feschber** [comment]
ah and I wanted to ask if I should mention https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/issues/126
since that was a bit frustrating to figure out...

File: src/desktop/input_capture.rs (hunk 5 above)
Comment (bilelmoussaoui) [reply src/desktop/input_capture.rs:221]: Ah indeed, my bad.

**Comment by bilelmoussaoui** [comment]
> ah and I wanted to ask if I should mention https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/issues/126 since that was a bit frustrating to figure out...

that would be nice, while keeping it clear that it only happens with the gnome implementation

**Comment by bilelmoussaoui** [comment]
Thanks a lot!

**Pull request merged**